core-services/src/main/java/org/zalando/nakadi/partitioning/PartitionResolver.java

```diff
@@ -68,7 +69,21 @@ public String resolvePartition(final EventType eventType, final JSONObject event
 
         final List<String> partitions = timelineService.getTopicRepository(eventType)
                 .listPartitionNames(timelineService.getActiveTimeline(eventType).getTopic());
-        return partitionStrategy.calculatePartition(eventType, eventAsJson, partitions);
+        return partitionStrategy.calculatePartition(eventType, PartitionData.fromJson(eventType, eventAsJson), partitions);
     }
 
+    public String resolvePartition(final EventType eventType, final GenericRecord eventAsBinary) {
```

Comment:
I would rather pass as an argument PartitionData to avoid code duplication, and from the calling method create PartitionData